can't build with gradle

Hi, trying to get the perspective examples going with the instructions here:
https://github.com/inductiveautomation/ignition-sdk-examples/tree/master/perspective-component

I've installed everything, and run  ./gradlew buildModule ( ./gradlew.sh doesn't exist). 

I get the following: 

`Downloading https://services.gradle.org/distributions/gradle-5.2.1-all.zip
..............................................................................................................................

Welcome to Gradle 5.2.1!

Here are the highlights of this release:
 - Define sets of dependencies that work together with Java Platform plugin
 - New C++ plugins with dependency management built-in
 - New C++ project types for gradle init
 - Service injection into plugins and project extensions

For more details see https://docs.gradle.org/5.2.1/release-notes.html

Starting a Gradle Daemon (subsequent builds will be faster)

> Task :collectToModlDeps
Copying toModl dependencies to perspective-component/build/libs

> Task :web:npmSetup
/home/max/ignition-sdk-examples/perspective-component/web/.gradle/npm/npm-v6.4.1/bin/npm -> /home/max/ignition-sdk-examples/perspective-component/web/.gradle/npm/npm-v6.4.1/lib/node_modules/npm/bin/npm-cli.js
/home/max/ignition-sdk-examples/perspective-component/web/.gradle/npm/npm-v6.4.1/bin/npx -> /home/max/ignition-sdk-examples/perspective-component/web/.gradle/npm/npm-v6.4.1/lib/node_modules/npm/bin/npx-cli.js
+ npm@6.4.1
added 387 packages from 770 contributors in 7.16s

┌───────────────────────────────────────────────────────┐
│                npm update check failed                │
│          Try running with sudo or get access          │
│         to the local update config store via          │
│ sudo chown -R $USER:$(id -gn $USER) /home/max/.config │
└───────────────────────────────────────────────────────┘

> Task :web:yarnSetup
/home/max/ignition-sdk-examples/perspective-component/web/.gradle/yarn/yarn-v1.15.2/bin/yarnpkg -> /home/max/ignition-sdk-examples/perspective-component/web/.gradle/yarn/yarn-v1.15.2/lib/node_modules/yarn/bin/yarn.js
/home/max/ignition-sdk-examples/perspective-component/web/.gradle/yarn/yarn-v1.15.2/bin/yarn -> /home/max/ignition-sdk-examples/perspective-component/web/.gradle/yarn/yarn-v1.15.2/lib/node_modules/yarn/bin/yarn.js
+ yarn@1.15.2
added 1 package in 0.521s

┌───────────────────────────────────────────────────────┐
│                npm update check failed                │
│          Try running with sudo or get access          │
│         to the local update config store via          │
│ sudo chown -R $USER:$(id -gn $USER) /home/max/.config │
└───────────────────────────────────────────────────────┘

> Task :web:yarn FAILED

FAILURE: Build failed with an exception.

* What went wrong:
A problem was found with the configuration of task ':web:yarn'.
> File '/home/max/ignition-sdk-examples/perspective-component/web/yarn.lock' specified for property '$2' does not exist.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 6.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/5.2.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 1m 2s
8 actionable tasks: 8 executed`

--- a/perspective-component/build.gradle
+++ b/perspective-component/build.gradle
@@ -13,7 +13,7 @@ buildscript {
 
     }
 
-    ext.sdk_version = "8.0.9"
+    ext.sdk_version = "8.0.13-SNAPSHOT"
 
     dependencies {
         classpath("com.inductiveautomation.gradle:ignition-module-plugin:1.2.10-SNAPSHOT")
@@ -44,7 +44,7 @@ allprojects {
         moduleId "org.fakester.radcomponent"
         moduleVersion "${project.version}"
         moduleDescription "A module that adds components to the Perspective module."
-        requiredIgnitionVersion "8.0.9"
+        requiredIgnitionVersion "8.0.13"
         requiredFrameworkVersion "8"
         isFree true
         license "license.html"

--- a/perspective-component/designer/build.gradle
+++ b/perspective-component/designer/build.gradle
@@ -9,6 +9,7 @@ targetCompatibility = JavaVersion.VERSION_11
 
 dependencies {
     compile(project(":common"))
+    compileOnly("com.google.code.findbugs:jsr305:3.0.1")
     compileOnly("com.inductiveautomation.ignitionsdk:designer-api:${sdk_version}")
     compileOnly("com.inductiveautomation.ignitionsdk:perspective-common:${sdk_version}")
     compileOnly("com.inductiveautomation.ignitionsdk:perspective-designer:${sdk_version}")

--- a/perspective-component/web/build.gradle
+++ b/perspective-component/web/build.gradle
@@ -9,8 +9,8 @@ plugins {
 // configurations on which versions of Node, Npm, and Yarn the gradle build should use.  Configuration provided by/to
 // the gradle node plugin that"s applied above (com.moowork.node)
 node {
-    version = "8.15.0"
-    yarnVersion = "1.15.2"
+    version = "10.18.1"
+    yarnVersion = "1.22.4"
     npmVersion = "6.4.1"
     download = true
     nodeModulesDir = file(project.projectDir)
@@ -20,23 +20,21 @@ node {
 String  mountedJsDir= "$project.parent.projectDir/gateway/src/main/resources/mounted"
 
 // define a gradle task that will install our npm dependencies, extends the YarnTask provided by the node gradle plugin
-
-
 task yarnPackages(type: YarnTask) {
 
     // group only used to categorize the task when running "gradle tasks" and viewing the list of possible tasks.group = "Web Build tasks"
     description = "Executes 'yarn' at the root of the web/ directory to install npm dependencies for the yarn workspace."
     // which yarn command to execute
-    args = ["install"]
+    args = ["install", "--verbose"]
 
     // set this tasks "inputs" to be any package.json files or yarn.lock files that are found in or below our current
     // folder (project.projectDir).  This lets the build system avoid repeatedly trying to reinstall dependencies
     // which have already been installed.  If changes to package.json or yarn.lock are detected, then it will execute
     // the install task again.
-    inputs.files(
-            fileTree(project.projectDir).matching {
-                include("**/package.json", "**/yarn.lock")
-            })
+//    inputs.files(
+//            fileTree(project.projectDir).matching {
+//                include("**/package.json")//, "**/yarn.lock")
+//            })
 
     dependsOn("${project.path}:yarn")
 }
@@ -64,6 +62,8 @@ task webpack(type: NpmTask) {
     // inputs.
     outputs.files(fileTree("${project.parent?.projectDir}/gateway/src/main/resources/mounted"))
 }
+
+project.tasks.findByName("build").dependsOn(webpack)
 
 // task to delete the dist folders
 task deleteDistFolders(type: Delete) {

--- a/perspective-component/web/package.json
+++ b/perspective-component/web/package.json
@@ -17,6 +17,7 @@
   "devDependencies": {
     "lerna": "3.13.1",
     "mini-css-extract-plugin": "^0.7.0",
+    "typescript": "3.6.5",
     "yarn": "1.22.4"
   },
   "resolutions": {

--- a/perspective-component/web/package.json
+++ b/perspective-component/web/package.json
@@ -17,7 +17,7 @@
   "devDependencies": {
     "lerna": "3.13.1",
     "mini-css-extract-plugin": "^0.7.0",
-    "yarn": "1.22.0"
+    "yarn": "1.22.4"
   },
   "resolutions": {
     "@types/react": "16.8.8",

--- a/perspective-component/web/packages/client/package.json
+++ b/perspective-component/web/packages/client/package.json
@@ -14,10 +14,10 @@
         "deepClean": "npm run clean && rimraf node_modules __coverage__"
     },
     "dependencies": {
-        "@inductiveautomation/perspective-client": "^1.0.3",
+        "@inductiveautomation/perspective-client": "1.0.13",
         "axios": "^0.19.0",
-        "mobx": "5.9.0",
-        "mobx-react": "5.4.3",
+        "mobx": "5.15.1",
+        "mobx-react": "6.1.8",
         "react": "16.8.4",
         "react-dom": "16.8.4"
     },
@@ -35,10 +35,10 @@
         "sass-loader": "^7.1.0",
         "ts-loader": "5.3.3",
         "tslint": "5.14.0",
-        "tslint-react": "^3.6.0",
-        "typescript": "3.3.4000",
+        "tslint-react": "3.6.0",
+        "typescript": "3.6.4",
         "webpack": "4.29.6",
         "webpack-cli": "3.3.0",
-        "yarn": "^1.17.3"
+        "yarn": "^1.22.4"
     }
 }

--- a/perspective-component/web/packages/client/tsconfig.json
+++ b/perspective-component/web/packages/client/tsconfig.json
@@ -6,8 +6,7 @@
         "baseUrl": ".",
         "paths" : {
           "*": [
-              "*",
-              "../../node_modules/"
+              "*"
           ]
         },
         "declaration": true,

--- a/perspective-component/web/packages/client/typescript/components/Messenger.tsx
+++ b/perspective-component/web/packages/client/typescript/components/Messenger.tsx
@@ -22,8 +22,8 @@
  *
  */
 
-import { observable, action, reaction, computed } from 'mobx';
-import { Disposer, observer } from 'mobx-react';
+import { observable, action, reaction, computed, IReactionDisposer } from 'mobx';
+import { observer } from 'mobx-react';
 import * as React from 'react';
 import {
     AbstractUIElementStore,
@@ -99,7 +99,7 @@ export class CustomValueStore implements MessageResponseHandler {
      * this is called when a component is removed from the DOM by calling the disposer in React's
      * 'componentWillUnmount()' lifecycle is called.
      */
-    messagePendingDisposer?: Disposer;
+    messagePendingDisposer?: IReactionDisposer;
 
     /**
      * Observable value that is set when a message should be pending.  Setting this to 'true' will trigger a 'reaction'

--- a/perspective-component/web/packages/designer/package.json
+++ b/perspective-component/web/packages/designer/package.json
@@ -17,8 +17,8 @@
         "@inductiveautomation/perspective-client": "1.0.13",
         "@inductiveautomation/perspective-designer": "1.0.13",
         "@myCompanyNamespace/example-client": "1.0.0",
-        "mobx": "5.15.1",
         "bind-decorator": "^1.0.11",
+        "mobx": "5.15.1",
         "react": "^16.8.4",
         "react-dom": "^16.8.4"
     },
@@ -28,6 +28,7 @@
         "@types/react-dom": "^16.8.3",
         "ts-loader": "5.3.3",
         "tslint": "5.14.0",
+        "tslint-react": "^5.0.0",
         "typescript": "3.6.4",
         "webpack": "4.29.6",
         "webpack-cli": "3.3.0",

--- a/perspective-component/web/packages/designer/package.json
+++ b/perspective-component/web/packages/designer/package.json
@@ -14,9 +14,10 @@
         "deepClean": "npm run clean && rimraf node_modules __coverage__"
     },
     "dependencies": {
-        "@inductiveautomation/perspective-client": "^1.0.3",
-        "@inductiveautomation/perspective-designer": "^1.0.3",
+        "@inductiveautomation/perspective-client": "1.0.13",
+        "@inductiveautomation/perspective-designer": "1.0.13",
         "@myCompanyNamespace/example-client": "1.0.0",
+        "mobx": "5.15.1",
         "bind-decorator": "^1.0.11",
         "react": "^16.8.4",
         "react-dom": "^16.8.4"
@@ -27,9 +28,9 @@
         "@types/react-dom": "^16.8.3",
         "ts-loader": "5.3.3",
         "tslint": "5.14.0",
-        "typescript": "3.3.4000",
+        "typescript": "3.6.4",
         "webpack": "4.29.6",
         "webpack-cli": "3.3.0",
-        "yarn": "^1.17.3"
+        "yarn": "1.22.4"
     }
 }

--- a/perspective-component/web/packages/designer/tsconfig.json
+++ b/perspective-component/web/packages/designer/tsconfig.json
@@ -6,8 +6,7 @@
         "baseUrl": ".",
         "paths" : {
           "*": [
-              "*",
-              "../../node_modules/"
+              "*"
           ]
         },
         "declaration": true,

